Fix/corl 2480 correct open story drawer

marcushaddon opened the pull request:
## What does this PR do?
Removes the StoryActions dropdown and replaces uses that button to open the StoryInfoDrawer, removes "Open in new" icon in StoryRow

## What changes to the GraphQL/Database Schema does this PR introduce?
None.

## Does this PR introduce any new environment variables or feature flags?
No.

## If any indexes were added, were they added to INDEXES.md?
n/a

## How do I test this PR?
Go to Admin > Stories, see that the "Open in. new" icon is gone. Click the three dots that used to open StoryActions and see that the Story Info Drawer opens

marcushaddon committed:
remove metadata

marcushaddon committed:
implement changes from Coral Party

marcushaddon committed:
update submit button text

marcushaddon committed:
remove eslint disable

marcushaddon committed:
address comments

marcushaddon committed:
fix tests

marcushaddon committed:
add post shadow dom TODO

marcushaddon committed:
hide config options if story archived
show spinner if viewer is null while rendering moderate story button
add todos for after https://github.com/coralproject/talk/pull/3796 is merged

marcushaddon committed:
remove unused type

marcushaddon committed:
replace StoryActions with StoryInfoDrawer

marcushaddon committed:
make open drawer icon a button
localize aria label

marcushaddon committed:
WIP bring Archive/Unarchive logic from StoryActions

marcushaddon committed:
hook up button

marcushaddon committed:
resolve conflicts

marcushaddon committed:
show archive story button if not archived

marcushaddon committed:
fix snapshot
fix linting
fix some styling

marcushaddon committed:
temporarily skip test of action menu

nick-funk committed:
Rename ArchiveStory to ArchiveStoryActionsContainer

nick-funk reviewed (approved):
Looks good, and my testing shows it works, approved!

nick-funk committed:
Test as much of the story drawer as we can

Also add notes as to why we can't get a hold of the
story drawer elements to test them further.

nick-funk commented:
Put a don't merge label on as I tried to update the tests. I managed to fix them up somewhat, but there is a lacking ability to access the modal div within the `testRenderer` so I left some notes. This at least tests the opening of the story info drawer though.

Removed the label, going to let tests pass and merge this into develop now.

nick-funk committed:
Remove non-existent viewer prop from StoryTable
